subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -82,6 +82,19 @@ Use other collections or I/O frameworks instead of `org.gradle.util.CollectionUt
 
 Gradle plugin authors may find the Designing Gradle Plugins subsection on link:https://guides.gradle.org/designing-gradle-plugins/#restricting_the_plugin_implementation_to_gradle_s_public_api[restricting the plugin implementation to Gradle's public API] helpful.
 
+[[sec:declaring_tasks]]
+=== Declaring tasks in a build script
+
+The task API gives a build author a lot of flexibility to declare tasks in a build script.
+For optimal readability and maintainability follow these rules:
+
+* The task type should be the only key-value pair that belongs within the parentheses after the task name.
```

Comment:
This isn't really true for the Kotlin DSL. Also... why?

Reply:
Regarding Kotlin...this is a whole different beast. We'll have to change so many things across the user guide that I wouldn't consider this needed at the moment. We won't be able to "just" offer two different sample snippets, one in Groovy and one in Kotlin. The surrounding text will have to explain and support.

Maybe we have to tackle this with the Kotlin DSL documentation work and not do this right now. However, the statements listed are important for the Groovy DSL.

Reply:
I think some of the statements are important, but this doesn't seem that bad:

```
task myTask(type: MyTask, 
                      dependsOn: otherTask, 
                      description: 'This does something', 
                      group: 'build') {}
```

The only thing I can say is objectively better about your recommendation here is that it makes the declaration more consistent with Kotlin DSL (where, I think, one must use methods)

Reply:
What it really boils down to is inconsistent code styles and therefore less readability and consistency. I have seen projects that use your notation and my proposed notation in different places. It's horrible. If I could I would completely remove the `Map` parameter from the API. In most cases, I think it's not a good idea to allow users different notation to achieve the exact same thing.

Reply:
I like that we are being opinionated here since we cannot enforce this on an API level, so I'll allow this.